### ISSUE TITLE
Approximate Training Time

### ISSUE THREAD

**Issue by ii35322** [Langboat/mengzi-retrieval-lm#8]
Hello, 
Thanks for the authors to have this repo, it's really helpful to me. Now I'm using this repo to training with my dataset and my customize database. I use 8 V-100 GPUs and the utilization rate is of each GPU is nearly 100%. However, the training time is extremely slow, only 1 epoch per day. If I train the GPT-Neo-125M without retro (just use huggingface), It can train 40 epochs per day. So I want to ask that is there has any bottleneck to make the training become much slower in the retrieval process? or may I ask that how long did you train the retro model and get the result of this repo? Thanks!

**Comment by bling0830** [comment]
Are you retrieving while training, and your retrieval service and model training are deployed on the same GPU?

**Comment by ii35322** [comment]
Yes, but I use 8 V-100 GPUs, and I set 6 encoder actors. Only one GPU has both retrieval service and model training. The other 7 GPUs only be used for model training.

**Comment by bling0830** [comment]
How many indexes do you use for retrieval and how many corpora does your dataset have? We recommend deploying retrieval service and model training separately.

If you don't mind, we are curious about what tasks you are using the retro for.

**Comment by ii35322** [comment]
Now I use 1 gpu as a retrieval service and the other 7 gpus to deploy the model training, the speed is much more reasonable! Thanks for the explanation!

**Comment by bling0830** [comment]
In order to train faster and improve GPU usage, we recommend preparing training corpus with retrieval information in advance, [prepare_load.py](https://github.com/Langboat/mengzi-retrieval-lm/blob/main/train/prepare_load.py) can be used as a reference code for batch preparation.

**Comment by ii35322** [comment]
That is very useful advice, I'll change the chunk size to 32 or 16 and see what's going on. And I'll also take look the prepare_load.py file to see whether I can speed up the training. Thanks! Also, is this repo can change the backbone model? E.g., switch got-neo to gpt2, or gpt2-medium?

**Comment by saisurbehera** [comment]
Hello @ii35322 

Are you using the 16GB or 32GB variant for the V100? I have OOM errors on the 16GB variant.

**Comment by ii35322** [comment]
Hello, @saisurbehera 

I use 32GB for the V100, if you need to use 16GB, I think you can reduce the batch size or the max length in mengzi-retrieval-lm/train/config.json

**Comment by saisurbehera** [comment]
@ii35322, How much did the final thing in training for all the epochs ?
